doc/pantry.md

```diff
@@ -0,0 +1,482 @@
+<div class="hidden-warning"><a href="https://docs.haskellstack.org/"><img src="https://rawgit.com/commercialhaskell/stack/master/doc/img/hidden-warning.svg"></a></div>
+
+# Pantry in Stack
+
+Beginning with Stack 1.11, Stack uses the Pantry library for its
+specification of snapshots and package locations. Under the surface,
+Pantry is geared towards reproducible build plans with
+cryptographically secure specification of packages and snapshots.
```

Comment:
If version changes, it needs to change here, too (there are a few more places in the file.